router/router.go

```diff
@@ -488,15 +488,13 @@ func (router *Router) emitSystemFunctionInvocationFailed(space string, functionI
 		return
 	}
 
-	if _, ok := err.(*function.ErrFunctionError); ok {
-		system := eventpkg.New(
-			eventpkg.SystemFunctionInvocationFailedType,
-			mimeJSON,
-			eventpkg.SystemFunctionInvocationFailedData{Space: space, FunctionID: functionID, Event: event, Error: err})
-		router.handleAsyncSubscriptions(http.MethodPost, systemPathFromSpace(space), *system, nil)
+	system := eventpkg.New(
+		eventpkg.SystemFunctionInvocationFailedType,
+		mimeJSON,
+		eventpkg.SystemFunctionInvocationFailedData{Space: space, FunctionID: functionID, Event: event, Error: err})
```

Comment:
This function will only get triggered if the `emit` itself failed, right? So we don't need to confirm the `err != nil` before this right?

Reply:
This function is called only if the error is nil.